pkg/detector/eventfilter.go

```diff
@@ -0,0 +1,28 @@
+package detector
+
+import (
+	"reflect"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/klog/v2"
+)
+
+// SpecificationChanged check if the specification of the given object change or not
+func SpecificationChanged(oldObj, newObj *unstructured.Unstructured) bool {
+	oldBackup := oldObj.DeepCopy()
+	newBackup := newObj.DeepCopy()
+
+	// Remove the status and some system defined mutable fields in metadata, including managedFields and resourceVersion.
+	// Refer to https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/object-meta/#ObjectMeta for more details.
+	removedFields := [][]string{{"status"}, {"metadata", "managedFields"}, {"metadata", "resourceVersion"}}
+	for _, r := range removedFields {
+		unstructured.RemoveNestedField(oldBackup.Object, r...)
+		unstructured.RemoveNestedField(newBackup.Object, r...)
+	}
+
+	if reflect.DeepEqual(oldBackup, newBackup) {
+		klog.V(4).Infof("Ignore update event of object (kind=%s, %s/%s) as specification no change", oldBackup.GetKind(), oldBackup.GetNamespace(), oldBackup.GetName())
```

Comment:
We don't need to print this log in this function, we can move this log to the place where use it. 
Since from this function of view, it is not sure it will be used in `update event`.